File: Practical3.java
public class Practical3 {
  public static void main(String[] args) {
    functions stack = new functions(10);
    stack.push(1);
    stack.push(2);
    stack.push(3);
    stack.push(4);
    stack.push(5);
    System.out.println(stack.pop());
    stack.display();
    System.out.println("Value of 2 element from top: " + stack.peep(2));
    stack.Change(3, 12);
    stack.display();
    functions obj = new functions(10);
    obj.display();
  }
}
class functions {
  int Top = -1;
  Integer size = null;
  Integer[] S;

  public functions(int size) {
    this.size = Integer.valueOf(size);
    this.S = new Integer[this.size];
  }
  Boolean push(Integer value) {
    if (this.Top >= S.length) {
      System.out.println("Overflow");
      return false;
    }
    this.Top++;
    S[Top] = value;
    return true;
  }
  Integer pop() {
    if (this.Top == -1) {
      System.out.println("Underflow");
      return null;
     }
    Integer temp = this.S[this.Top];
    this.S[this.Top] = null;
    this.Top--;
    return temp;
  }
  Integer peep(int index) {
    if ((index - 1) <= (this.Top) && this.Top - (index - 1) > -1) {
      return this.S[this.Top - (index - 1)];
    } else {
      System.out.println("Invalid index");
      return null;
    }
  }
  Boolean Change(int index, Integer value) {
    if (this.Top == -1) {
      System.out.println("UnderFlow");
      return false;
    }
    if (this.Top < index) {
      System.out.println("Invalid Index");
      return false;
    }
    this.S[index] = value;
    return true;
  }
  void display() {
    if (this.Top == -1) {
      System.out.println("underflow");
      return;
    }
    System.out.print("[");
    for (Integer i : this.S) {
      System.out.print(i + " ");
    }
    System.out.println("]");
  }
}

File: Practical3.py
class Stack:
    stack = []

    def __init__(self, size):
        self.size = size

    def getStack(self):
        return self.stack

    def push(self, value):
        if not len(self.stack) >= self.size:
            self.stack.append(value)
            return True
        else:
            print("Stack Overflowed!!")
            return False

    def pop(self):
        if not len(self.stack) == 0:
            return self.stack.pop()
        else:
            print("Stack UnderFlowed!!")
            return False

    def peep(self, index):
        if len(self.stack)-index >= 0:
            return self.stack[int(f"-{index}")]
        else:
            return False

    def change(self, index, value):
        if not index >= len(self.stack):
            self.stack[index] = value
            return True
        else:
            return False

    def display(self):
        if len(self.stack) == 0:
            print("Underflow!!")
            return;
        for i in range(0, self.size):
            if i < len(self.stack):
                print(self.stack[i], end=" ")
            else:
                print("none", end=" ")
        print("")


size = int(input("Enter the size of the stack: "))
obj = Stack(size)
obj.push(5)
obj.push(4)
obj.push(3)
obj.push(2)
obj.push(1)
obj.push(0)  # won't be pushed as stack is full
obj.display()
print("I am popped: ", obj.pop())
print("i am popped: ", obj.pop())
obj.display()
print("Second element from top: ", obj.peep(2))
obj.push(2)
obj.push(1)
obj.change(3, 45)
obj.display()
mystack = obj.getStack()
print(mystack)
